### ISSUE TITLE
Custom predefined container definitions

### ISSUE THREAD

**Issue by bossm8** [microsoft/vscode-remote-release#5653]
I will explain the feature request based on an example we got in our company:

We have our own LaTeX classes and templates which contain information we are not allowed to make public (so adding them to the public repo is no option). We installed these in private docker images to automate builds in e.g. pipelines. 
To develop in vscode remote containers we need to add the exact same .devconfig.json to each project in order to use the container. Also the configuration "remote.containers.repositoryConfigurationPaths": [] is not really an option since it needs the repository mirrored (#3712), so we would need to make a new directory structure for each project, which not really an optimization.

Long story short. It would be nice to have a configuration option where one could specify a local or remote repo where some devcontainer configuration similar to [vscode-dev-containers](https://github.com/microsoft/vscode-dev-containers) are located which are then available like the predefined container definitions. Maybe in a section called "Custom predefined container definitions". So everyone in our team could just configure the repo and have the devcontainers available without always configuring.

**Comment by bossm8** [comment]
Marking this my own as a duplicate of [#4852](https://github.com/microsoft/vscode-remote-release/issues/4852).
